include/boost/graph/undirected_dfs.hpp

```diff
@@ -78,10 +78,10 @@ namespace detail
                 if (v_color == Color::white())
                 {
                     vis.tree_edge(*ei, g);
-                    src_e = *ei;
                     stack.push_back(std::make_pair(u,
-                        std::make_pair(src_e, std::make_pair(++ei, ei_end))));
+                        std::make_pair(src_e, std::make_pair(ei+1, ei_end))));
```

Comment:
Probably has to be this so that it works with edge iterators that aren't random access.
```suggestion
                        std::make_pair(src_e, std::make_pair(std::next(ei), ei_end))));
```
And, call me fussy, but I prefer this style anyway.

Reply:
Ah yes, makes sense! I'm not very fluent in C++, so thanks for pointing this out :)